### PR TITLE
Feature/add locale support

### PR DESCRIPTION
**Pull request by tomerlevi444**
- Reads locale from application's configuration, if defined.

### CODE REVIEW

**Comment by tomerlevi444** [comment]
@grozen @yuval-netanel Please review.

**Comment by yuval-netanel** [comment]
Aprroved

File: klarna-on-demand-android.iml [hunk 1]
```diff
@@ -1,21 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (grozen) [inline klarna-on-demand-android.iml:1]: Where'd this file go? Is this one of those damn IDE files that keep popping in and out?

File: klarna-on-demand/src/main/java/com/klarna/ondemand/Context.java [hunk 2]
```diff
@@ -24,4 +26,10 @@ protected static String getApiKey(){
         }
         return apiKey;
     }
+
+    static String getLocale(android.content.Context context) {
+        Locale locale = context.getResources().getConfiguration().locale;
+
+        return locale.getLanguage();
```
Comment (grozen) [inline klarna-on-demand/src/main/java/com/klarna/ondemand/Context.java:33]: So are we getting a locale or a language here? What does this return?

File: klarna-on-demand/src/main/java/com/klarna/ondemand/UrlHelper.java [hunk 3]
```diff
@@ -50,12 +50,8 @@ static String preferencesUrl(String token) {
                 .appendPath(token)
                 .appendPath("preferences")
                 .appendQueryParameter("api_key", Context.getApiKey())
-                .appendQueryParameter("locale", defaultLocale())
+                .appendQueryParameter("locale", Context.getLocale(context))
```
Comment (grozen) [inline klarna-on-demand/src/main/java/com/klarna/ondemand/UrlHelper.java]: I'm sure you can tell me what I'm missing, but this construct (`Context.getLocale(context)`) is quite awkward. Is there no way around this?

File: klarna-on-demand-android.iml (hunk 1 above)
Comment (tomerlevi444) [inline klarna-on-demand-android.iml:1]: Yes

File: klarna-on-demand/src/main/java/com/klarna/ondemand/UrlHelper.java (hunk 3 above)
Comment (tomerlevi444) [inline klarna-on-demand/src/main/java/com/klarna/ondemand/UrlHelper.java]: changed to applicationContext

File: klarna-on-demand/src/main/java/com/klarna/ondemand/Context.java (hunk 2 above)
Comment (tomerlevi444) [inline klarna-on-demand/src/main/java/com/klarna/ondemand/Context.java:33]: changed to getLanguate.
Thanks!

**Comment by grozen** [comment]
Approved. What's up with those failing tests, though?

**Pull request merged**